src/servers/src/http/handler.rs

```diff
@@ -65,6 +70,102 @@ pub struct SqlQuery {
     pub limit: Option<usize>,
 }
 
+#[derive(Debug, Default, Serialize, Deserialize, JsonSchema)]
+pub struct LogIngesterQueryParams {
+    pub table_name: String,
+}
+
+fn validate_log_ingester_payload(payload: &Value) -> Result<(), String> {
+    if !payload.is_object() {
+        return Err("payload must be an object".to_string());
+    }
+
+    if payload["processors"].as_str().is_none() {
+        return Err("processors field is required".to_string());
+    }
+
+    if payload["data"].as_array().is_none() {
+        return Err("data field is required".to_string());
+    }
+
+    Ok(())
+}
+
+/// handler to log ingester
+#[axum_macros::debug_handler]
+pub async fn log_ingester(
+    State(_state): State<LogHandlerRef>,
+    Query(_query_params): Query<LogIngesterQueryParams>,
+    Extension(_query_ctx): Extension<QueryContextRef>,
+    Json(mut _payload): Json<Value>,
+) -> HttpResponse {
+    // TODO (qtang): implement log ingester
+    // transform payload to RowInsertRequest
+    // maybe we need a new trait for log ingester like `LogIngester` instead of `ServerGrpcQueryHandlerRef`
+    // let request = pileline::transform(payload);
+    // state.do_query(payload, query_ctx).await.to_string();
+    match validate_log_ingester_payload(&_payload) {
+        Ok(_) => (),
+        Err(e) => {
+            return HttpResponse::Error(ErrorResponse::from_error_message(
+                StatusCode::InvalidArguments,
+                e,
+            ))
+        }
+    };
+    let processors_ = _payload["processors"].take();
+    let processors = processors_.as_str().unwrap();
+    let data = _payload["data"].take();
+    let yaml_content = Content::Yaml(processors.into());
+    let pipeline_: Result<Pipeline<GreptimeTransformer>, String> = parse(&yaml_content);
+    match pipeline_ {
+        Ok(pipeline) => {
+            let pipeline_data = PipelineValue::try_from(data);
+            match pipeline_data {
+                Ok(pipeline_data) => {
+                    let transformed_data: Result<Rows, String> = pipeline.exec(pipeline_data);
+                    match transformed_data {
+                        Ok(rows) => {
```

Comment:
Should we validate the `rows` in advance, for example, there must be a `time index` column